StitchCoreServicesAwsS3-iOS/StitchCoreServicesAwsS3-iOS/Internal/AwsS3ServiceClientImpl.swift

```diff
@@ -0,0 +1,139 @@
+import Foundation
+import MongoSwift
+import StitchCore
+import StitchCore_iOS
+import StitchCoreServicesAwsS3
+
+public final class AwsS3ServiceClientImpl: AwsS3ServiceClient {
+    private let proxy: CoreAwsS3ServiceClient
+    private let dispatcher: OperationDispatcher
+    
+    internal init(withClient client: CoreAwsS3ServiceClient,
+                  withDispatcher dispatcher: OperationDispatcher) {
+        self.proxy = client
+        self.dispatcher = dispatcher
+    }
+    
+    public func putObject(bucket: String,
+                          key: String,
+                          acl: String,
+                          contentType: String,
+                          body: String,
+                          _ completionHandler: @escaping (AwsS3PutObjectResult?, Error?) -> Void) {
+        self.dispatcher.run(withCompletionHandler: completionHandler) {
+            return try self.proxy.putObject(
+                bucket: bucket,
+                key: key,
+                acl: acl,
+                contentType: contentType,
+                body: body
+            )
+        }
+    }
+    
+    public func putObject(bucket: String,
+                          key: String,
+                          acl: String,
+                          contentType: String,
+                          body: String,
+                          timeout: TimeInterval,
```

Comment:
[nit] should use default args for timeout, if possible

Reply:
as we discussed, this is not possible because protocols can't have default parameters